ST_environs.c

```diff
@@ -191,73 +234,73 @@ static void _make_disturbance( void) {
 
 /*------------------------------------------------------*/
   /* Can't have simultaneous disturbances*/
-  if (Plot.disturbance != NoDisturb) {
-    switch( Plot.disturbance) {
+  if (Plot->disturbance != NoDisturb) {
+    switch( Plot->disturbance) {
       case FecalPat:
-           if (Plot.pat_removed) {
-             Plot.disturbed = 0;
-             Plot.pat_removed = FALSE;
-             Plot.disturbance = NoDisturb;
+           if (Plot->pat_removed) {
+             Plot->disturbed = 0;
+             Plot->pat_removed = FALSE;
+             Plot->disturbance = NoDisturb;
            } else {
-             pc = Globals.pat.recol[Slope] * Plot.disturbed
-                  + Globals.pat.recol[Intcpt];
+             pc = Globals->pat.recol[Slope] * Plot->disturbed
+                  + Globals->pat.recol[Intcpt];
              if (RandUni(&environs_rng) <= pc) {
-               Plot.pat_removed = TRUE;
+               Plot->pat_removed = TRUE;
                /* slight effects for one year*/
-               Plot.disturbed = 1;
+               Plot->disturbed = 1;
              } else {
-               Plot.pat_removed = FALSE;
-               Plot.disturbed++;
+               Plot->pat_removed = FALSE;
+               Plot->disturbed++;
              }
            }
            break;
       case NoDisturb: // Does nothing but prevent a compiler warning
       case LastDisturb:
       default:
-           Plot.disturbed = (Plot.disturbed) ? Plot.disturbed -1 : 0;
+           Plot->disturbed = (Plot->disturbed) ? Plot->disturbed -1 : 0;
     }
-    if (Plot.disturbed == 0)
-      Plot.disturbance = NoDisturb;
+    if (Plot->disturbed == 0)
+      Plot->disturbance = NoDisturb;
 
   }
 
   /* if the disturbance was expired above, */
   /* we can generate a new one immediately */
-  if (Plot.disturbance == NoDisturb) {
+  if (Plot->disturbance == NoDisturb) {
 
     /* pick some type of disturbance (other than none)*/
     event = (DisturbEvent) RandUniIntRange(1, LastDisturb -1, &environs_rng);
 
     /* make sure this is off unless needed  */
-    Plot.pat_removed = FALSE;
+    Plot->pat_removed = FALSE;
     switch( event) {
       case FecalPat:
-       if (!Globals.pat.use) {event=NoDisturb; break;}
-         event = (RandUni(&environs_rng) <= Globals.pat.occur)
+       if (!Globals->pat.use) {event=NoDisturb; break;}
+         event = (RandUni(&environs_rng) <= Globals->pat.occur)
                ? event : NoDisturb;
          if (event == NoDisturb) break;
-         Plot.pat_removed = (RandUni(&environs_rng) <= Globals.pat.removal)
+         Plot->pat_removed = (RandUni(&environs_rng) <= Globals->pat.removal)
                            ? TRUE : FALSE;
```

Comment:
Why use a ternary here when you can just use the boolean expression itself?

Reply:
I've seen quite a few things like this in the code. Good catch!

Reply:
Resolved by 9b99edd.